code/game/objects/items/devices/laserpointer.dm

```diff
@@ -69,7 +69,7 @@
 	laser_act(target, user, params)
 
 /obj/item/laser_pointer/proc/laser_act(atom/target, mob/living/user, params)
-	if( !(user in (viewers(7,target))) )
+	if( !(user in (viewers(7,target))) || !(target in view(user.loc)) )

```

Comment:
`view()` is really expensive.

Reply:
is `if( !(target in view(7,user.loc)) )` less expensive?

Reply:
No. Just use `can_see()`.